docs/dev-client-networking.md

```diff
@@ -3,259 +3,647 @@ id: dev-client-networking
+Networked entities are any entities with the `Networked` component. The
+`Networked` component defined in
+[`src/bit-components.js`](https://github.com/mozilla/hubs/blob/master/src/bit-components.js)
+contains:
+
+- A (networked) `id`, which clients use to uniquely identify this entity across
+the network. This cannot simply be the `eid` of an entity, because `eid`s are
+assigned locally to each client.
+- A `creator`, which is used at various times to determine whether or not an
+entity should be created or removed.
+- An `owner`, which is used to determine which `client` has authority to update
+the state of this entity.
+- A `lastOwnerTime`, which is used to determine the most recent time that an
+`owner` was assigned.
+- A `timestamp`, which is the most recent time the networked state of this
+entity has been updated.
+
+The `creator` can be set to a `ClientId`, `"reticulum"`, or a `NetworkID`.
+
+- When the `creator` is a `ClientID`, it means that a client in the room has
+caused this &ldquo;root&rdquo; entity (and its descendants) to be created, and
+the entity (and its descendants) should be removed when the client leaves the
+room.
+- When the `creator` is `"reticulum"`, it means that the Reticulum is
+responsible for deciding whether this &ldquo;root&rdquo; entity should be
+created or removed.
+- When the `creator` is a `NetworkID`, it means that the entity is a descendant
+of a &ldquo;root&rdquo; entity, and its creation/removal will be subject to its
+ancestor.
+
+Conceptually, the `creator` and the `owner` act as authorities over two facets
+of a networked entity.
+
+- The `creator` is the authority over the entity&rsquo;s existence. Thus, it is
+checked when processing `CreateMessage` s and before an entity may be removed.
+For an entity to be created, its `creator` must have the appropriate
+permission. An entity&rsquo;s `creator` changes infrequently. It only happens
+when a client `pins` (or `unpins`) an entity, which changes the `creator` to
+(or from) `"reticulum"`.
+- The `owner` is the authority over the entity&rsquo;s current state. Thus, it
+is associated with `UpdateMessage`s. The `owner` is expected to change
+regularly, whenever a new client performs an action on an entity. The
+`takeOwnership` and `takeSoftOwnership` functions allow a client to establish
+itself as the `owner` of an entity.
+
+
+### Message Types
+
+Clients send and receive these types of messages:
+
+- `CreateMessage`
+- `UpdateMessage`
+- `DeleteMessage`
+- `ClientJoin`
+- `ClientLeave`
+- `CreatorChange`
+
+Note that this is a slight simplification. The message types are not
+represented in these exact terms throughout the client. For example, clients
+may combine `CreateMessage`s, `UpdateMessage`s, and `DeleteMessage`s into a
+single outgoing message, which receiving clients then separate and parse. There
+are also two variants of `UpdateMessage`, which will be explained below.

```

Comment:
This note adds a lot of confusion for me. I find myself wanting more information but it doesn’t feel correct to go into detail here because we haven’t learned about the simplest uses of these Messages yet. Perhaps this could go at the bottom of the “Message Types” section with a small example.

Reply:
Same above.

Reply:
Reflected a comment in #160, thanks.